sensor/kubernetes/eventpipeline/output/output_impl.go

```diff
@@ -0,0 +1,70 @@
+package output
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/sensor/common/detector"
+	"github.com/stackrox/rox/sensor/kubernetes/eventpipeline/component"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type outputQueueImpl struct {
+	innerQueue   chan *component.ResourceEvent
+	stopSig      *concurrency.Signal
+	forwardQueue chan *central.MsgFromSensor
+	detector     detector.Detector
+}
+
+// Send sends a ResourceEvent message to the inner queue
+func (q *outputQueueImpl) Send(msg *component.ResourceEvent) {
+	q.innerQueue <- msg
+}
+
+// ResponsesC returns the MsgFromSensor channel
+func (q *outputQueueImpl) ResponsesC() <-chan *central.MsgFromSensor {
+	return q.forwardQueue
+}
+
+// Start starts the outputQueueImpl component
+func (q *outputQueueImpl) Start() error {
+	go q.runOutputQueue()
+	return nil
+}
+
+// Stop stops the outputQueueImpl component
+func (q *outputQueueImpl) Stop(_ error) {
+	defer close(q.innerQueue)
+	defer close(q.forwardQueue)
+	q.stopSig.Signal()
+}
+
+// runOutputQueue reads messages from the inner queue, forwards them to the forwardQueue channel
+// and sends the deployments (if needed) to Detector
+func (q *outputQueueImpl) runOutputQueue() {
+	for {
+		select {
+		case <-q.stopSig.Done():
+			return
+		case msg, more := <-q.innerQueue:
```

Comment:
Related to the message above, it seems redundant to have both a `case` for `stropSig` being triggered and handling closed channels. 

Maybe a simpler way to code this is to drop the `select` block entirely, since we can rely on `Stop()` being called and channels being closed.

```go
func (q *outputQueueImpl) runOutputQueue() {
	for {
                msg, more := <-q.innerQueue
                if !more {
                       return
                }
                // ...
        }
}
```
Reading on closed channels will return `(nil, true)` if there are no more values buffered. 

For now we can do without the signal and I think the simplification in the code is justified. I'm fine to introduce it back again in case we want to propagate output errors.

Reply:
The only problem that I see there is: If we call `Stop()` and there are 10 events buffered in the channel, we won’t stop until all of them are processed (if I'm not mistaken). Wouldn’t it make sense to just stop processing at the time Stop is called instead of processing the 10 events and then stopping?

Reply:
Yes, you're correct. That's a fair point. I think the `Stop` and `ErrorSignal` path can distinguish the graceful stop and crashing.